test/extended/builds/image_source.go

```diff
@@ -251,3 +205,22 @@ var _ = g.Describe("[Feature:Builds][Slow] build can have Docker image source",
 		})
 	})
 })
+
+func checkBuildEnv(value string) {
+	obj, _, err := legacyscheme.Codecs.UniversalDecoder().Decode([]byte(value), nil, nil)
+	o.Expect(err).NotTo(o.HaveOccurred())
+	ok := false
+	build, ok := obj.(*buildapi.Build)
+	o.Expect(ok).To(o.BeTrue(), "could not convert build env\n %s\n to a build object", value)
+
+	o.Expect(build.Spec.Strategy.CustomStrategy.From.Kind).To(o.Equal("DockerImage"))
+	o.Expect(build.Spec.Strategy.CustomStrategy.From.Name).To(o.ContainSubstring("@sha256:"))
+	o.Expect(build.Spec.Strategy.CustomStrategy.PullSecret).NotTo(o.BeNil())
```

Comment:
each test used a different field to check (build.Spec.Strategy.DockerStrategy vs SourceStrategy vs. CustomStrategy), so this won't work as a generic utility function.